### PR TITLE
Task/chain

### PR DESCRIPTION
**Pull request by birdca**
Take 4 Chains in sample:
1. LLMChain
2. SimpleSequentialChain
3. SequentialChain
4. Router Chain

### Result for LLMChain Sample
```
Choosing the best name for a software company that specializes in customized web front-end, back-end, and DevOps requires a name that is memorable, reflects the company's expertise, and ideally hints at the innovative and technical nature of the services offered. Here are five name suggestions that could fit such a company:

1. CodeCrafters Solutions: This name implies a high level of skill and attention to detail, likening software development to craftsmanship. The word "Solutions" indicates that the company provides effective resolutions to technical challenges.

2. DevSync Innovations: "DevSync" suggests a harmonious integration of development practices, which is crucial for combining front-end, back-end, and DevOps. "Innovations" positions the company as a forward-thinking leader in the industry.

3. NexusLogic Systems: "Nexus" denotes a connection or series of connections linking two or more things, which fits a company that integrates different aspects of software development. "Logic" emphasizes the company's focus on rational, systematic approaches, and "Systems" reinforces the idea of comprehensive solutions.

4. WebFusion Architects: This name conveys the idea of merging different web technologies to create robust solutions, much like an architect designs buildings. It implies a strategic approach to building software systems.

5. StackPioneer Tech: The term "StackPioneer" suggests leading-edge expertise in full-stack development, from front-end to back-end, along with DevOps. "Tech" makes it clear that the company is in the technology sector.

Remember to check for the availability of the desired name as a domain and as a registered business to ensure that you can establish a unique identity for your company. Additionally, it's always a good idea to conduct a trademark search to avoid any potential legal issues down the line.
```

### Result for  SimpleSequentialChain Sample
```
> Entering new SimpleSequentialChain chain...
Creating a name for a company is a creative process that should reflect not only the services it provides but also its brand identity, values, and vision. When naming a company that specializes in customized web front-end, back-end, and DevOps solutions, you may want to communicate aspects of innovation, reliability, and technical expertise. Here are five suggestions that could serve as a starting point for your company's name:

1. CodeCraft Solutions
   - Implying a masterful and artisanal approach to crafting software, this name suggests bespoke web development and operational excellence.

2. WebMatrix Innovations
   - This name conveys the comprehensive and interconnected services the company offers while emphasizing its innovative approaches to web development and DevOps practices.

3. DevSync Technologies
   - Highlighting the synchronization and integration between development and operations, this name suggests a seamless and efficient approach to custom software creation.

4. NexaForge Digital
   - 'Nexa' suggests a connection to the future or next advancements, combined with 'Forge,' which implies the creation and shaping of digital products with precision and strength.

5. TailorStack Systems
   - By incorporating the word 'Tailor,' this name emphasizes the company's focus on customized solutions, while 'Stack' indicates a full range of services from the front-end to the back-end, including DevOps. 

Make sure to perform a thorough search to ensure the names are not already trademarked or in use by another company in your industry. Additionally, consider the availability of domain names that match your company name for a cohesive online presence.
CodeCraft Solutions: Bespoke web development, operational excellence, artisanal software craftsmanship. 

WebMatrix Innovations: Comprehensive, interconnected, and innovative web and DevOps services.

DevSync Technologies: Seamless integration, efficient custom software, synchronized development and operations.

NexaForge Digital: Precision-shaped digital products, future-focused, innovative web solutions.

TailorStack Systems: Custom-tailored web solutions, full-service development, and operations stack.

> Finished chain.
CodeCraft Solutions: Bespoke web development, operational excellence, artisanal software craftsmanship. 

WebMatrix Innovations: Comprehensive, interconnected, and innovative web and DevOps services.

DevSync Technologies: Seamless integration, efficient custom software, synchronized development and operations.

NexaForge Digital: Precision-shaped digital products, future-focused, innovative web solutions.

TailorStack Systems: Custom-tailored web solutions, full-service development, and operations stack.
```

### Result for SequentialChain Sample
```
> Entering new SequentialChain chain...

> Finished chain.
{'Review': "Je trouve le goût médiocre. La mousse ne tient pas, c'est bizarre. J'achète les mêmes dans le commerce et le goût est bien meilleur...\nVieux lot ou contrefaçon !?", 'english_review': "I find the taste mediocre. The foam doesn't hold, it's weird. I buy the same ones in the store and the taste is much better...\nOld batch or counterfeit!?", 'summary': "The reviewer is disappointed with the product's lackluster taste and poor foam quality, suggesting it might be an old batch or counterfeit compared to the superior versions bought in-store.", 'language': 'The review is written in French.', 'followup_message': "Réponse:\n\nCher(e) client(e),\n\nNous sommes sincèrement désolés d'apprendre que votre expérience avec notre produit n'a pas répondu à vos attentes. La qualité et la satisfaction de nos clients sont nos priorités absolues et vos commentaires sont essentiels pour nous.\n\nConcernant votre déception par rapport au goût et à la qualité de la mousse, nous prenons vos préoccupations très au sérieux. Il est possible que vous ayez reçu un lot qui n'est pas à la hauteur de nos standards habituels ou, comme vous l'avez supposé, un produit qui n'est pas authentique.\n\nNous vous prions de bien vouloir nous contacter directement avec des détails supplémentaires, tels que le numéro de lot et le lieu d'achat, afin que nous puissions enquêter rapidement sur cette affaire. Si le produit que vous avez acheté est avéré être ancien ou contrefait, nous prendrons les mesures nécessaires pour résoudre le problème et veillerons à ce que vous receviez ce que vous attendez de notre marque.\n\nEn guise de compensation pour cette expérience décevante, nous aimerions vous offrir un bon de réduction pour votre prochain achat ou vous envoyer un remplacement gratuit du produit.\n\nNous vous remercions pour votre achat et espérons avoir la chance de restaurer votre confiance en notre marque.\n\nCordialement,\n\n[Votre Nom]\n[Poste]\n[Nom de la Société]"}
```

### Router Chain
While a simple sample may not suffice to demonstrate the complexity of this library, it is definitely worth exploring. I've included a note in the [documentation](https://www.notion.so/Note-for-Chains-in-Lanchain-b567de304c32483cae33f99798462421?pvs=4) for your reference.

### CODE REVIEW

File: llm/prompt/template/openai/chat.py [hunk 1]
```diff
@@ -38,3 +39,65 @@ def get_stylish_completion(prompt, style="Passionate and friendly"):
...
+    # chain 4: input= summary, language and output= followup_message
+    chain_four = LLMChain(
+        llm=llm, prompt=fourth_prompt, output_key="followup_message"
+    )
+
+    # and output= English_Review,summary, followup_message
+    overall_chain = SequentialChain(
+        chains=[chain_one, chain_two, chain_three, chain_four],
+        input_variables=["Review"],
+        output_variables=["English_Review", "summary","followup_message"],
```
Comment (timhuang1018) [inline llm/prompt/template/openai/chat.py]: I notice the response example with some keys showing capitals, seems like they are following by the keys provide here. We should consider align keys with all lowercase letters

File: llm/prompt/template/openai/chat.py [hunk 2]
```diff
@@ -38,3 +39,65 @@ def get_stylish_completion(prompt, style="Passionate and friendly"):
...
+
+def llmChain(prompt_input, chain_input):
+    llm = ChatOpenAI(temperature=0.9, model=llm_model)
+    prompt = ChatPromptTemplate.from_template(prompt_input)
+    chain = LLMChain(llm=llm, prompt=prompt)
+
+    return chain.run(chain_input)
+
+
+def simpleSequentialChain(prompt1, prompt2, chain_input):
```
Comment (timhuang1018) [inline llm/prompt/template/openai/chat.py]: consider using *args to make this function more reuable

File: llm/prompt/template/openai/chat.py [hunk 3]
```diff
@@ -38,3 +39,65 @@ def get_stylish_completion(prompt, style="Passionate and friendly"):
...
+    chain_two = LLMChain(llm=llm, prompt=second_prompt)
+
+    overall_simple_chain = SimpleSequentialChain(
+        chains=[chain_one, chain_two], verbose=True
+    )
+
+    return overall_simple_chain.run(chain_input)
+
+
+def sequentialChain(prompt1, prompt2, prompt3, prompt4, review):
```
Comment (timhuang1018) [inline llm/prompt/template/openai/chat.py]: maybe using **kwargs to make this function input with key and value. Keys like "summary", "reviews"... etc would better be passed from sample code(where this function be used). To make it more usable

File: llm/prompt/template/openai/chat.py (hunk 1 above)
Comment (birdca) [reply llm/prompt/template/openai/chat.py]: okay

File: llm/prompt/template/openai/chat.py (hunk 2 above)
Comment (birdca) [reply llm/prompt/template/openai/chat.py]: Great suggestion! I've reworked it using a list. Give it a look and tell me if it works for you. Thanks!

File: llm/prompt/template/openai/chat.py (hunk 3 above)
Comment (birdca) [reply llm/prompt/template/openai/chat.py]: Great suggestion! I've reworked it using a list. Give it a look and tell me if it works for you. Thanks!

**Review by timhuang1018 — APPROVED** [review]

**Pull request merged**